### ISSUE TITLE
Continue Training: -Clear flag

### ISSUE THREAD

**Issue by CAT1210** [AlexeyAB/darknet#8648]
I still have not completely wrapped my head around continuing training. I thought I grasped it fine after all my reading and I thought that I would give the -clear flag a try but that just led to more confusion.  I have an original dataset of 105K images (about 50% are negatives--don't have any objects I wish to detect). I trained my model through all 105K iterations. Then, by the time it finished training all that, I had another dataset all ready to add to the trained model from the 105K images.

So, using the last weights from the 105K images, I began training on only the new dataset (about 2000 images), using the -clear flag since I wanted to clear the epochs and clean slate. I let it train through 2000 iterations. When it finished, I ran some tests on some images, in this case, a few images of a squirrel. The original 105K trained model worked great, gave high 90% accuracy for all of them, as I would expect since the images were a part of the dataset. I ran the new model with the new dataset training and in only one image was a squirrel detected. It had high accuracy, which is great, but the other 7 images had no detection of a squirrel at all.

I am at a loss as to why this would happen. I understand that I am using the weight file from the 105K images as "pretrained" weights but I feel that would mean that the weights from that training would still apply but it looks like they were blown away after training my new dataset. Anybody have any ideas? In my new dataset, some objects I want to detect did not have any images, could this be why? Would I need new images from every object when I use the -clear flag? Is it because the new dataset was so small and not the 2000 for each object and that blew away the prior trained model?

I know that I can just continue training by raising the batches in cfg, but that seems to take much longer to roll through a new dataset than when I use the -clear flag...maybe that is an indication that something isn't right on my end...

**Comment by stephanecharette** [comment]
You cannot train with only the new images.  You must make sure you train with all images.  The only thing `-clear` does is remove the image counter at the head of the weights file.  That way you don't have to keep increasing your max batches.

**Comment by CAT1210** [comment]
Thank you so, so much for that info. Much clearer now.
